lib/python/compliance_checker_test.py

```diff
@@ -0,0 +1,162 @@
+    criteria = sub_collection[1]['check_params']['criteria']
+    if criteria == []:
+        criteria = "normal"
+    else:
+        criteria = criteria[0]
+
+    return {
+        'file_url': sub_collection[1]['path'][0],
+        'check_success_tests': sub_collection[1]['check_params']['checks_success'],
+        'check_fail_tests': sub_collection[1]['check_params']['checks_fail'],
+        'criteria': criteria,
+        'skip_checks': sub_collection[1]['check_params']['skip_checks']
+    }
+
+
+# collection is equivalent to a facility/sub-facility in the json file
+for collection in compliance_config:
+    print("Running test suite for: {collection}".format(collection=collection))
+
+    for sub_collection in compliance_config[collection].items():
+
+        def run_tests_netcdf(test_type):
+            """
+            run required tests on NetCDF. Amend results to the main sub_collection
+            dictionnary
+
+            test_type: authorized values 'check_success_tests', 'check_fail_tests'
+            """
+            # para_results_att value is an result attribute of the json file
+            if test_type == 'check_success_tests':
+                param_results_att = 'checks_success_tests_results'
+            elif test_type == 'check_fail_tests':
+                param_results_att = 'checks_fail_tests_results'
+            else:
+                raise ValueError("test_type: {test_type} not in ['check_success_tests' 'check_fail_tests']".format(test_type=test_type))
+
+            compliance_config[collection][sub_collection[0]][param_results_att] = {}
+
+            nc_filename = os.path.basename(info['file_url'])
+            print('  {nc_filename}'.format(nc_filename=nc_filename))
+
+            print('    {test_type}: {tests}'.format(test_type=test_type,
+                                                    tests=info[test_type]))
+
+            for test in info[test_type]:
+                try:
+                    res, keep_outfile_path = pass_netcdf_checker(
+                        tempfile_path, tests=[test],
+                        criteria=info['criteria'],
+                        skip_checks=info['skip_checks'],
+                        keep_outfile=True,
+                        output_format='text'
+                    )
+
+                except ValueError:
+                    print("compliance checker failed for \"{test}\" applied to {nc_filename}".format(
+                        test=test,
+                        nc_filename=nc_filename))
+
+                    res = False
+
+                """ In the case the test failed, the compliance output file is saved and moved to OUTPUT_DIR """
+                if res is False:
+                    err_filename = '{filename}_error_results.txt'.format(filename=nc_filename)
+                    compliance_config[collection][sub_collection[0]][param_results_att].\
+                        setdefault('{test}_failure_filename'.format(test=test), []).append(err_filename)
```

Comment:
Trying to make sense of this ties my brain into a knot... and I still have no idea.
Please try to break it up into a few lines to make it clearer!

Reply:
i renamed ```compliance_config[collection][sub_collection[0]][param_results_att]``` into a different variable for you. Dunno if that unknots anything?
but basically, if the compliance checker fails on a specific test and NetCDF, a line is added in the output json with the filename (not path) of the compliance checker result output, for example "IMOS_FAIMMS_T_20091217T023000Z_DAVSF1_FV01.nc_error_results.txt". This file is written in the temporary ouput folder alongside the json file